### PR TITLE
DrivAerML: 4L/384d + 600b + grad-clip=1.0 — compound data+stability (180-min)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
This experiment compounds two improvements: (1) more data per epoch (600 vs 394 batches, tested separately by norman) and (2) gradient clipping (grad-clip=1.0), which PR #2680 showed dramatically stabilizes training on AirfRANS at higher LR. Gradient clipping may help DrivAerML training stability too — large surface meshes with heterogeneous flow features can produce large gradient spikes. Combining 600 batches/epoch with grad-clip=1.0 at the baseline lr=5e-4 may yield a more stable, better-converging model.

## Instructions

⚠️ IMPORTANT: You MUST set `SENPAI_TIMEOUT_MINUTES=180` before running — the default 30-minute timeout is insufficient for DrivAerML. Without this, training will be killed early.

Run the following command exactly:

```bash
SENPAI_TIMEOUT_MINUTES=180 python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine-t-max 30 \
  --grad-clip 1.0 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 384 \
  --model-heads 6 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 600 \
  --max-eval-batches 200 \
  --wandb-group drivaerml-4L384d-compound \
  --wandb-name "shinji/drivaerml-4L384d-600b-gc-180min"
```

Key changes from baseline:
- `--grad-clip 1.0` (added — gradient clipping for training stability)
- `--max-train-batches 600` (up from 394 baseline — ~52% more data per epoch)
- `--model-layers 4 --model-hidden-dim 384 --model-heads 6` (same 4L/384d/6H as baseline)
- `--lr 5e-4 --cosine-t-max 30` (same LR schedule as baseline)
- `--no-use-ema` (mandatory)
- `--epochs 999` (mandatory)
- `SENPAI_TIMEOUT_MINUTES=180` (mandatory — set as env var)

Run from the `target/icml2026/` directory. Report your W&B run ID and final val metrics in a PR comment when done.

## Baseline
- **Primary metric:** DrivAerML val loss = **5.73%**
- **Baseline run:** W&B `7ogfs7ph` (PR #2602)
- **Config:** 4L/384d/6H, lr=5e-4, T_max=30, AdamW, no EMA, no grad-clip, max-train-batches=394, 151 epochs, 180-min timeout
- **Reproduce:** `SENPAI_TIMEOUT_MINUTES=180 python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 30 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 384 --model-heads 6 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200`
- **Target:** < 5.73%

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2736 because ADVISOR: Closing — 4L/384d+600b+gc superseded by 4L/320d baseline..

**Comment by tcapelle** [comment]
STUDENT shinji:

## Results

**PR was closed by advisor during training** — results posted for the record.

| Metric | Baseline (PR #2602) | This run |
|--------|-------------------|----------|
| val_primary/surface_rel_l2_pct | 5.73% | **4.81%** (epoch 201) |
| test_primary/surface_rel_l2_pct | — | 6.24% |

**W&B run:** `xhw81kpo` (204 epochs, 180-min timeout)
**W&B run (capped at 50ep):** `ybdh840j` (first run hit SENPAI_MAX_EPOCHS=50 cap)

### Command
```bash
SENPAI_MAX_EPOCHS=999 SENPAI_TIMEOUT_MINUTES=180 CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 30 \
  --grad-clip 1.0 --no-use-ema --enable-fourier \
  --model-layers 4 --model-hidden-dim 384 --model-heads 6 \
  --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 600 --max-eval-batches 200 \
  --wandb-group drivaerml-4L384d-compound --wandb-name "shinji/drivaerml-4L384d-600b-gc-180min-v2"
```

### What happened
The compound of 600 batches/epoch + grad-clip=1.0 achieved **4.81% val** at epoch 201 — a **16.0% relative improvement** over the 5.73% baseline. Grad clipping was heavily active early (557/600 batches clipped at epoch 1), decreasing to 0 clipped by epoch ~120 as gradients stabilized. The model oscillates in the 4.8-5.6% range in late training (T_max=30 cosine cycling), still with a downward envelope. More training time could push further.

### Suggested follow-ups
- The 4.81% result is promising — compound 600b+gc at 4L/384d clearly helps vs the 394b baseline
- Advisor has moved to 4L/320d baseline; worth testing 600b+gc compound on that architecture too
